### PR TITLE
feat(lsp): highlight active parameter in signature help

### PR DESCRIPTION
**Pull request by folke**
This PR adds highlighting with `LspSignatureActiveParameter` to signature help.

### CODE REVIEW

**Comment by mjlbach** [comment]
So I dug into this a bit based on our conversation on Matrix. Pyright (pyright 1.1.126) seems to be fine for me and is respecting the capabilities you added in 74527a3. I think the sumneko case is entirely server side and isolated (I checked, the return is incorrect and inconsistent with other servers), so approved.

**Review by mjlbach — APPROVED** [review]

File: runtime/lua/vim/lsp/handlers.lua [hunk 1]
```diff
@@ -328,6 +328,8 @@ M['textDocument/implementation'] = location_handler
 ---     - border:     (default=nil)
 ---         - Add borders to the floating window
 ---         - See |vim.api.nvim_open_win()|
+---     - active_parameter: (default=true)
```
Comment (mjlbach) [inline runtime/lua/vim/lsp/handlers.lua:331]: This doesn't look like it's used.

**Review by mjlbach — CHANGES_REQUESTED** [review]
Missed something, see comment

File: runtime/lua/vim/lsp/handlers.lua (hunk 1 above)
Comment (folke) [reply runtime/lua/vim/lsp/handlers.lua:331]: Missed that, done now :)

**Review by mjlbach — APPROVED** [review]

**Comment by justinmk** [comment]
does this mean https://github.com/neovim/neovim/pull/14456 can be closed?

**Comment by mjlbach** [comment]
Yes, we still need to fix some things (https://github.com/neovim/neovim/issues/15022) but went with this implementation.

**Pull request merged**